spec/state_of_the_nation_spec.rb

```diff
@@ -341,4 +341,35 @@ class Country < ActiveRecord::Base
       expect(washington.send(:should_round_timestamps?)).to eq(true)
     end
   end
+
+  context "IdentityCache support" do
+    before do
+      # Mock up of the IdentityCache methods
+      module IdentityCache
+        def fetch_presidents
+          self.presidents
+        end
+      end
+
+      class Country < ActiveRecord::Base
+        include StateOfTheNation
+        include IdentityCache 
+
+        has_many :presidents
+
+        has_uniquely_active(:president).with_identity_cache
+      end
+    end
+
+    let(:country) { Country.create }
+    let!(:president) { country.presidents.create!(entered_office_at: day(1), left_office_at: day(8)) }
+
+    context ".using_identity_cache" do
+      it "uses the fetch_method to retrieve records" do
+        expect(country).to receive(:fetch_presidents)
+          .and_call_original
+        expect(country.active_president(day(7))).to eq(president)
```

Comment:
this isn't a good test – if line 360 didn't include `with_identity_cache` it'd still pass